### PR TITLE
tab_switcher: Add tab close buttons

### PR DESCRIPTION
**Pull request by alygin**
Support for closing tabs from Tab Switcher:

- Close button color matches the indicator color to preserve the information that the buffer is dirty (as in SublimeText).
- `ctrl-backspace` closes the currently selected item.

https://github.com/zed-industries/zed/assets/2101250/8ea33911-2f62-4199-826d-c17556db8e9a

### CODE REVIEW

**Comment by mikayla-maki** [comment]
Nice! Is there a keyboard-accessible way to close tabs with this UI?

**Comment by alygin** [comment]
> Is there a keyboard-accessible way to close tabs with this UI?

No, there's not but I like the idea. I think I'll add `ctrl-backspase` as in IDEA. I didn't find such functionality in VSCode.

**Comment by mikayla-maki** [comment]
That sounds perfect!

**Comment by alygin** [comment]
Here it is.

**Comment by osiewicz** [comment]
Actually, I could use removing/"closing" items in tasks modal too; how do you feel about picker supporting that in it's interface?
I could also roll my own removal, just like you did. Maybe we won't have that many places that want to support the removal.

File: crates/tab_switcher/src/tab_switcher.rs [hunk 1]
```diff
@@ -266,14 +317,50 @@ impl PickerDelegate for TabSwitcherDelegate {
...
+            .into_any_element();
 
         Some(
             ListItem::new(ix)
                 .spacing(ListItemSpacing::Sparse)
                 .inset(true)
                 .selected(selected)
                 .child(h_flex().w_full().child(label))
-                .children(indicator),
+                .when(true, |el| {
```
Comment (osiewicz) [inline crates/tab_switcher/src/tab_switcher.rs:356]: This `when(true` smells a bit. Could we perhaps use `map` or `when_else` there? `map` in particular should be a bit less verbose.

Comment (alygin) [reply crates/tab_switcher/src/tab_switcher.rs:356]: Oh, sure, that looks bad.

**Comment by alygin** [comment]
> Actually, I could use removing/"closing" items in tasks modal too; how do you feel about picker supporting that in it's interface? I could also roll my own removal, just like you did. Maybe we won't have that many places that want to support the removal.

There's also the "Open recent project" that already supports items removing. But I'm not sure this feature can be easily generalized:
- Tab Switcher has this indicators and custom buttons coloring.
- Recent Projects Opener doesn't allow to close the current project.
- I assume the Tasks window will allow to remove any item.

Do you think all these cases should be supported by the Picker?

File: crates/tab_switcher/src/tab_switcher.rs (hunk 1 above)
Comment (alygin) [reply crates/tab_switcher/src/tab_switcher.rs]: Fixed

**Review by osiewicz — APPROVED** [review]
Hey,
The way you put it it indeed doesn't make sense to generalize deletion to be an inherent property of a picker.

Given that I think this PR looks good to be merged, thanks :)

P.S: Sorry for the late response, but it's a national holiday in Poland.

File: crates/picker/src/picker.rs [hunk 2]
```diff
@@ -49,6 +49,9 @@ pub trait PickerDelegate: Sized + 'static {
     fn set_selected_index(&mut self, ix: usize, cx: &mut ViewContext<Picker<Self>>);
 
     fn placeholder_text(&self, _cx: &mut WindowContext) -> Arc<str>;
+    fn no_matches_text(&self, _cx: &mut WindowContext) -> Arc<str> {
```
Comment (osiewicz) [inline crates/picker/src/picker.rs:52]: nit: this could return a SharedString instead of `Arc<str>`. Feel free to ignore it though.

Comment (alygin) [reply crates/picker/src/picker.rs]: Fixed

**Comment by alygin** [comment]
> Sorry for the late response, but it's a national holiday in Poland.

No worries at all! Thanks for the review!

**Pull request merged**